### PR TITLE
[HIGH] Update CI/CD for docs

### LINKED ISSUE #73: Update and test documentation CD
Continuous deployment (CD) of the documentation should be reactivated.
One can use the [BIG-MAP/LabNotebookAppOntology](https://github.com/BIG-MAP/LabNotebookAppOntology) as a template for how it should be done.

### PR DESCRIPTION
**Pull request by CasperWA**
~Update documentation CI/CD workflows similarly as to what is done in [BIG-MAP/LabNotebookAppOntology](https://github.com/BIG-MAP/LabNotebookAppOntolog).~

Closes #73 

**Updated/Reworked this PR**:

**_IMPORTANT_**: A change in this PR adds `.ttl` to the end of the `owl:versionIRI` value. If this is not intended, this should be reverted. One thing is that the IRI will currently not be resolvable due to the `battinfo.ttl` file being released under an extra sub-folder for v0.6.0. This can be changed manually, simultaneously with merging this PR, while a new version should proabably be released as well (v0.6.1?). How to proceed on this should be agreed upon before merging this PR and it should be well coordinated to ensure no real down time for users of BattINFO.

Following @eibar-flores's excellent work with updating the documentation build workflow, this builds on that and ensures the ontology is also "published" as intended.
The CI/CD workflows have been updated and expanded in the following way:

- The CI tests workflow now properly tests building the documentation as well as "publishing" the ontologies/-y.
- The Update GH-Pages workflow will now push to the `gh-pages` branch, updating it with the latest documentation and ontologies as well as preserving earlier versions of the ontologies.
  Furthermore, it's made callable by other workflows, useful for the "release" workflow (mentioned below).
- A new CD Update Version workflow, or "release" workflow, has been added, which will run when a GitHub release is published. It will take the new tag and use it to update the `owl:versionIRI` and `owl:versionInfo` values in the ontologies (now only `battinfo.ttl`) and also it's reference in the `catalog-v001.xml` file. Then it will merge those changes back to `master` and run the Update GH-Pages workflow, effectively publishing the new ontology version.

Finally, a bit of developer documentation has been added to explain the intended way of publishing the documentation as well as releasing a new BattINFO version.

### CODE REVIEW

**Comment by CasperWA** [comment]
@jesper-friis and @jsimonclark do you know what the error is here? ([Failing action run](https://github.com/BIG-MAP/BattINFO/runs/5394988249?check_suite_focus=true))

**Comment by jesper-friis** [comment]
I think `ChargePerMassDimension` has been changed to `ElectricChargePerMassDimension`. Could it be that the test is using an old source?

**Review by jesper-friis — COMMENTED** [review]
The changes looks good. But we should resolve the failing test before merge

**Comment by CasperWA** [comment]
> I think `ChargePerMassDimension` has been changed to `ElectricChargePerMassDimension`. I guess that the reason for the failing test is that source `/home/runner/work/BattINFO/BattINFO/doc/ontodoc/tmp/battinfo-inferred.ttl` is outdated.

Is this something that either you @jesper-friis or @jsimonclark can fix directly in the branch for this PR? :)

**Comment by CasperWA** [comment]
Updated this completely to comply with the excellent documentation updates implemented by @eibar-flores.

Tests now succeed - the actual publishing workflow now needs to be fixed and remedied.

File: docs/scripts/md_to_html.py [hunk 1]
```diff
@@ -1,84 +1,116 @@
...
 
+    if not output_path.is_absolute():
+        # Relative path - Must be relative to the repository root
+        output_path = (repo_dir / output_path).resolve()
 
-########### RUN THE RENDERING WORKFLOW ##############
-
-def rendering_workflow():
+    # Get all Markdown files NOT in the assets, css, or scripts directories
+    md_files = [
```
Comment (eibar-flores) [inline docs/scripts/md_to_html.py:68]: I dont fully follow the code from lines 63 to 85. I think you are scanning through md files not in the assets, css and scripts, but you have already moved them there. What is the purpose of this part of the code?

**Comment by eibar-flores** [comment]
Could you elaborate on what are the consequences of having (or not) the ttl ending on the `owl:versionIRI` field?

**Comment by eibar-flores** [comment]
Greatly appreciated the two `**.md` files added to the docs directory, providing useful information for developers. We should perhaps consider publishing these as part of a "For developers" tab in the deployed documentation :)

**Comment by jsimonclark** [comment]
we've updated the documentation workflow according to recommendations from the EMMO working group.

**Pull request closed without merging**